testdata/main.go

```diff
@@ -6,32 +6,122 @@ import (
 	"time"
 	"context"
 	"strings"
+	"os"
+	"encoding/json"
 
 	a "github.com/ibmdb/go_ibm_db"
 )
 
 var ctx = context.Background()
 
-var host = "<HOST>"
-var port = "<PORT>"
-var database = "<DATABASE>"
-var uid = "<UID>"
-var pwd = "<PWD>"
+var host string
+var port string
+var database string
+var uid string
+var pwd string
+var connStr string
 
-var connStr = "PROTOCOL=tcpip;HOSTNAME=" + host + ";PORT=" + port + ";DATABASE=" + database + ";UID=" + uid + ";PWD=" + pwd
+//Read Config variable from json file
+type Config  struct {
+        Host string `json:"HOSTNAME"`
+        Port string `json:"PORT"`
+	Database string `json:"DATABASE"`
+	Uid string `json:"UID"`
+	Pwd string `json:"PWD"`
+}
+
+func LoadConfiguration(filename string) (Config, error) {
+        var config Config
+        configFile, err := os.Open(filename)
+        defer configFile.Close()
+        if err != nil {
+                return config, err
+        }
+        jsonParser := json.NewDecoder(configFile)
+        err = jsonParser.Decode(&config)
+        return config, err
+}
+
+//Get connection information from config.json
+func GetConnectionInfoFromConfigFile() {
+       config, _:= LoadConfiguration("config.json")
+       host = config.Host
+       port = config.Port
+       database = config.Database
+       uid = config.Uid
+       pwd =  config.Pwd
+}
+
+
+//Get connection information from environment variables 
+func GetConnectionInfoFromEnv() bool{
+	var databaseFound bool
+	var hostFound bool
+	var portFound bool
+	var uidFound bool
+	var pwdFound bool
+	var flag1 bool = false
+	database, databaseFound = os.LookupEnv("DB2_DATABASE")
+        if !databaseFound{
+		fmt.Println("Warning: Environment variable DB2_DATABASE is not set.")
```

Comment:
@vmathur12 Print warning only if env vars DB2_USER and DB2_PASSWD is not set. No need to set all connection string info from env vars only. Few can be read from config.json file too like SECURITY, SSLServerCertificate and other keywords including dbname, host, port.